core/state/statedb.go

```diff
@@ -144,13 +143,6 @@ func New(root common.Hash, db Database, snaps *snapshot.Tree) (*StateDB, error)
 		accessList:          newAccessList(),
 		hasher:              crypto.NewKeccakState(),
 	}
-	if sdb.snaps != nil {
```

Comment:
It seems as if snaps are necessary for the reorg logic at a certain depth, where has this been re-implemented?